### PR TITLE
AY-6740 CSV ingest: make Version column optional.

### LINKED ISSUE #16: AY-6740_CSV ingest: version column optional
### Please describe the enhancement you have in mind and explain what the current shortcomings are?

Client have been discussing AOV publishing a bit recently, so this might be a potential workflow for now. But their preference would be to develop a new publishing creator for Tray publisher for passes. But it feels like this CSV workflow isn't far off what I was asking for.

Current problem is that version column is required / hardcoded. This might be couple lines change to enhance the workflow for next available version.

### How would you imagine the implementation of the enhancemenent?
_No response_

### Describe alternatives you've considered:
_No response_

### Additional context:
()
_(might be a private channel)_

_This issue was automatically created from Clickup ticket [AY-6740](https://app.clickup.com/t/86c0ae8dk)_

### PR DESCRIPTION
**Pull request by robin-ynput**
## Changelog Description

These changes resolves #16 
Ensure the Version column in the CSV is optional:
* _When it exists and it's filled up,_ will publish and enforce provided version (**previous behavior**)
* _When it exists but it's empty,_ product will take next available version
* _When it does not exist,_ default logic applies, product will take next available version

## Testing notes:

Tested via the command line and interactive mode using:
* [AY_240319_0001_ABS - ay_240319_0001_abs_paths_60_empty.csv](https://github.com/user-attachments/files/17298103/AY_240319_0001_ABS.-.ay_240319_0001_abs_paths_60_empty.csv)  column exists but empty
* [AY_240319_0001_ABS - ay_240319_0001_abs_paths_60_mixed.csv](https://github.com/user-attachments/files/17298104/AY_240319_0001_ABS.-.ay_240319_0001_abs_paths_60_mixed.csv)  mix of version number and empty
* [AY_240319_0001_ABS - ay_240319_0001_abs_paths_no_version.csv](https://github.com/user-attachments/files/17298105/AY_240319_0001_ABS.-.ay_240319_0001_abs_paths_no_version.csv) version column missing

[How to use the CSV CLI interface](https://ayon.ynput.io/docs/addon_traypublisher_admin/#cli-interface)

1. Create the relevant shot hierarchy in the project
2. Ensure CSV settings are properly set and OCIO config is enabled
3. Download [resource material](https://github.com/ynput/ayon-core/files/14651928/ay_240319_0001.zip) and edit CSV to point toward downloading paths
4. Run the CSV ingest via CLI or interactive mode (traypublisher)

### CODE REVIEW

File: server/settings/creator_plugins.py [hunk 1]
```diff
@@ -215,8 +215,8 @@ class TrayPublisherCreatePluginsModel(BaseSettingsModel):
                 {
                     "name": "Version",
                     "type": "number",
-                    "default": "1",
-                    "required_column": True,
+                    "default": "0",
```
Comment (BigRoy) [inline server/settings/creator_plugins.py:218]: Hmm yeah. Tricky. Because technically version 0 is allowed to be a valid version.
I believe the setting for that is `ayon+settings://core/version_start_category`.

File: server/settings/creator_plugins.py [hunk 2]
```diff
@@ -182,7 +182,7 @@ class TrayPublisherCreatePluginsModel(BaseSettingsModel):
                     "type": "text",
                     "default": "",
                     "required_column": True,
-                    "validation_pattern": "^([a-zA-Z\\:\\ 0-9#._\\\\/]*)$"
+                    "validation_pattern": "^([a-zA-Z\\:\\ 0-9#-._\\\\/]*)$"
```
Comment (BigRoy) [inline server/settings/creator_plugins.py:185]: This fixes so that `-` is allowed in input filenames?

Comment (BigRoy) [reply server/settings/creator_plugins.py:185]: Actually, shouldn't `-` need to be escaped within `[]` for regex? (Using a backslash?) Otherwise it identifies a range, right? Like `[A-Za-z]`

Comment (BigRoy) [reply server/settings/creator_plugins.py:185]: Or put the hyphen last or first in the brackets, like explained [here](https://stackoverflow.com/a/3697210)

Comment (robin-ynput) [reply server/settings/creator_plugins.py]: Yeah, you're right. It worked on my side but I think it was a pure coincidence that `9#-.` interpreted as the range between `#` and `.` covers`-`. Also took the opportunity to adjust `.` to be `\\.` as well as my understanding is that we want to allow points, not any single character.

File: server/settings/creator_plugins.py (hunk 1 above)
Comment (robin-ynput) [reply server/settings/creator_plugins.py:218]: Yeah I was inspired this change from https://github.com/ynput/ayon-traypublisher/blob/develop/client/ayon_traypublisher/plugins/create/create_csv_ingest.py#L58 to get `None` as default instead of `1`.

Comment (BigRoy) [reply server/settings/creator_plugins.py:218]: Might be an edge case for now - this works fine with the default settings.

**Review by jakubjezek001 — APPROVED** [review]
Unfortunately I could not test it at the moment. But the code seems to be fine and @BigRoy tests were also passing, so perfect work!
